### PR TITLE
[15.0][MIG] sale_order_line_input: Migration to 15.0

### PR DESCRIPTION
**Pull request by cesar-tecnativa**
@Tecnativa TT36521

### CODE REVIEW

**Comment by cesar-tecnativa** [comment]
@CarlosRoca13 could you check?

**Review by CarlosRoca13 — APPROVED** [review]

**Comment by cesar-tecnativa** [comment]
@pedrobaeza could you check please?

File: sale_order_line_input/models/sale_order.py [hunk 1]
```diff
@@ -11,45 +11,45 @@ class SaleOrderLine(models.Model):
         related="order_id.pricelist_id",
         readonly=True,
     )
-    force_company_id = fields.Many2one(
+    with_company_id = fields.Many2one(
```
Comment (pedrobaeza) [inline sale_order_line_input/models/sale_order.py]: You don't need to rename this field.

File: sale_order_line_input/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        product, taxes..."""
+        for line in self:
+            line.company_id = line.with_company_id
+
+    @api.onchange("order_partner_id")
+    def _onchange_order_partner_id(self):
+        """Create order to correct compute of taxes"""
+        if not self.order_partner_id or self.order_id:
+            return
+        SaleOrder = self.with_context(with_company=self.with_company_id.id).env[
```
Comment (pedrobaeza) [inline sale_order_line_input/models/sale_order.py]: https://www.odoo.com/documentation/15.0/es/developer/howtos/company.html

```suggestion
        SaleOrder = self.with_company(self.force_company_id).env[
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: sale_order_line_input/models/sale_order.py [hunk 3]
```diff
@@ -27,7 +27,7 @@ def _compute_force_company_id(self):
             line.force_company_id = (
                 line.order_id.company_id
                 # Is not necessary use browse here
-                or self.env.context.get("force_company")
+                or self.env.context.get("with_company")
```
Comment (pedrobaeza) [inline sale_order_line_input/models/sale_order.py]: I think you have to look into `allowed_company_ids` context variable.

File: sale_order_line_input/i18n/es.po [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+#. module: sale_order_line_input
+#: model_terms:ir.ui.view,arch_db:sale_order_line_input.view_sales_order_line_filter
+msgid "Not Invoice Related"
+msgstr "Sin factura asociada"
+
+#. module: sale_order_line_input
+#: model_terms:ir.ui.view,arch_db:sale_order_line_input.view_sales_order_line_input_tree
+#, fuzzy
+msgid "Open Sale Order"
+msgstr "Línea de pedido de venta"
```
Comment (carlosdauden) [inline sale_order_line_input/i18n/es.po]: Bad translation

File: sale_order_line_input/views/sale_order_line_view.xml [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+                    required="0"
+                    placeholder="New"
+                    domain="[('state', 'not in', ('done', 'cancel')), ('company_id', '=', company_id)]"
+                    attrs="{'readonly': [('order_id', '!=', False)]}"
+                    force_save="1"
+                    options='{"no_open": True, "no_create": True}'
+                />
+                <button
+                    name="action_sale_order_form"
+                    string="Open Sale Order"
```
Comment (carlosdauden) [inline sale_order_line_input/views/sale_order_line_view.xml]: Remove this to avoid waste space

**Review by carlosdauden — CHANGES_REQUESTED** [review]

**Comment by victoralmau** [comment]
Could you review it again?

**Review by sergio-teruel — APPROVED** [review]

**Comment by sergio-teruel** [comment]
@victoralmau Add yourself as contributor

**Comment by sergio-teruel** [comment]
ping @carlosdauden

**Comment by carlosdauden** [comment]
Attend this:
https://github.com/OCA/sale-workflow/pull/2114#discussion_r930477479

**Comment by victoralmau** [comment]
Changes done.

**Comment by sergio-teruel** [comment]
@pedrobaeza review, if all is ok, Can you merge it?

**Review by carlosdauden — APPROVED** [review]

File: sale_order_line_input/models/sale_order.py (hunk 3 above)
Comment (pedrobaeza) [reply sale_order_line_input/models/sale_order.py]: This is not correct afaik. You should check allowed companies

File: sale_order_line_input/models/sale_order.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    )
+
+    @api.depends("order_id")
+    def _compute_force_company_id(self):
+        """Related company is not computed already when we click create new line"""
+        for line in self:
+            line.force_company_id = (
+                line.order_id.company_id
+                # Is not necessary use browse here
+                or self.env.context.get("with_company")
```
Comment (carlosdauden) [inline sale_order_line_input/models/sale_order.py]: Don't change original line:
```suggestion
                or self.env.context.get("force_company")
```

Comment (victoralmau) [reply sale_order_line_input/models/sale_order.py]: Change done.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**